nuxeo-features/rest-api/nuxeo-rest-api-server/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/directory/DirectoryEntryObject.java

```diff
@@ -101,7 +102,7 @@ private void checkDeleteGuards() {
         if (deleteConstraints != null && !deleteConstraints.isEmpty()) {
             for (DirectoryDeleteConstraint deleteConstraint : deleteConstraints) {
                 if (!deleteConstraint.canDelete(directoryService, entryId)) {
-                    throw new NuxeoException("This entry is referenced in another vocabulary.", SC_BAD_REQUEST);
+                    throw new NuxeoException("This entry is referenced in another vocabulary.", SC_CONFLICT);
```

Comment:
I guess you now need a PR on https://github.com/nuxeo/nuxeo-web-ui/blob/master/elements/directory/nuxeo-vocabulary-management.js#L321 to handle the `409` https SC?